src/pages/transactions/components/TransactionMatchDetails.tsx

```diff
@@ -131,6 +175,66 @@ export function TransactionMatchDetails(props: Props) {
     }
   };
 
+  const linkToExpense = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!expenseIds.length || isFormBusy) {
+      return;
+    } else {
+      setIsFormBusy(true);
+
+      toast.processing();
+
+      request('POST', endpoint('/api/v1/bank_transactions/match'), {
+        transactions: [
+          {
+            id: props.transactionDetails.transaction_id,
+            expense_id: expenseIds.join(','),
+          },
+        ],
+      })
+        .then(
+          (response: GenericSingleResourceResponse<TransactionResponse[]>) => {
+            queryClient.invalidateQueries('/api/v1/bank_transactions');
+            queryClient.invalidateQueries('/api/v1/expenses');
+            queryClient.invalidateQueries(
+              route('/api/v1/bank_transactions/:id', {
+                id: props.transactionDetails.transaction_id,
+              })
+            );
+            queryClient.invalidateQueries(
+              route('/api/v1/expenses/:id', {
+                id: response.data.data[0].expense_id,
+              })
+            );
+
+            toast.success('linked_transaction');
+          }
+        )
+        .catch((error: AxiosError) => {
+          console.error(error);
+          toast.error();
+        })
+        .finally(() => setIsFormBusy(false));
+    }
+  };
+
+  const getMatchingFunction = () => {
+    if (isCreate) {
+      if (props.isCreditTransactionType) {
+        return convertToPayment;
+      } else {
+        return convertToExpense;
+      }
+    } else {
+      if (props.isCreditTransactionType) {
+        return linkToPayment;
+      } else {
+        return linkToExpense;
+      }
+    }
+  };
```

Comment:
Is it possible to simplify this? Looks like a ton of nesting. Perhaps we can simplify it with early return?

Something like this should work:

```ts
const getMatchingFunction = () => {
  if (isCreate) {
    if (props.isCreditTransactionType) {
      return convertToPayment;
    }

    return convertToExpense;
  }

  if (props.isCreditTransactionType) {
    return linkToPayment;
  }

  return linkToExpense;
};
```

Reply:
Yeah, you are right. It's simplified.